crates/rune-modules/src/http.rs

```diff
@@ -284,6 +305,32 @@ impl RequestBuilder {
             request: self.request.body(bytes.into_std()),
         }
     }
+
+    #[rune::function]
+    fn form(self, params: HashMap<String, String>) -> Self {
```

Comment:
Rune HashMap; I don't see an issue with the conversion here.

Reply:
:thinking:

To avoid copying it, you can use `rune::modules::collection::hash_map::HashMap`, but you'd have to make the methods you need to use `pub` (which I'm fine with) but then you'd have to ensure it implements `serde::Serialize`.

If you want to support anonymous objects though, this should be a reference to `rune::runtime::Object`, then calling this would support this syntax:

```rust
let request = /* .. */;

request = request.form(#{field: "value"});
```

I think that already implements `Serialize`.

Reply:
Future work